applications/plugins/Communication/components/serverCommunicationVRPN.h

```diff
@@ -0,0 +1,68 @@
+#include "serverCommunication.h"
+
+#include<vrpn_Analog.h>
+#include<vrpn_Button.h>
+#include<vrpn_Tracker.h>
+#include<vrpn_Text.h>
+#include<vrpn_Connection.h>
+#include<vrpn_Configure.h>
+
+#define MAX 1024
+
+namespace sofa
+{
+
+namespace component
+{
+
+namespace communication
+{
+
+class SOFA_COMMUNICATION_API ServerCommunicationVRPN : public ServerCommunication
+{
+public:
+
+    typedef ServerCommunication Inherited;
+    SOFA_CLASS(ServerCommunicationVRPN, Inherited);
+
+    ServerCommunicationVRPN();
+    virtual ~ServerCommunicationVRPN();
+
+    //ArgumentList stringToArgumentList(std::string dataString);
+
+    //////////////////////////////// Factory VRPN type /////////////////////////////////
+    typedef CommunicationDataFactory VRPNDataFactory;
+    VRPNDataFactory* getFactoryInstance();
+    virtual void initTypeFactory() override;
+    /////////////////////////////////////////////////////////////////////////////////
+
+    virtual std::string getArgumentType(std::string value) override;
+    virtual std::string getArgumentValue(std::string value) override;
+
+protected:
+    vrpn_Connection *m_connection;
+    vrpn_Text_Sender *vrpn_text_sender;
+    vrpn_Analog_Server *vrpn_analog_server;
+    vrpn_Tracker_Server *vrpn_tracker_server;
+    timeval delay;
+
+    float angle = 0;
+    vrpn_float64 pos[3], d_quat[4];
+    const vrpn_uint32 class_of_service = vrpn_CONNECTION_LOW_LATENCY;
+    int sensor = 0;
```

Comment:
not sure if those variables have to be there. Can you confirm it ?

Reply:
Yes, they are used as parameter inside tracker `report_pos()` method.

Reply:
Then, move them inside this function :)

Reply:
Okay sure :)